### DISCUSSION TITLE
Lots of issues using CLI but no issues when using Jar Bazel

### DISCUSSION THREAD

**Discussion by manwinder123** [detekt/detekt#4888] [General]
## Expected Behavior
When I run `detekt` command in terminal it returns a bunch of issues, such as `ImplicitDefaultLocale` but when I am running `detekt` using the library `io.gitlab.arturbosch.detekt.api` I get 0 issues. 

## Steps to Reproduce
Here is my `ProcessingSpec`:
```
return ProcessingSpec {
        logging {
            debug = debugMode
            outputChannel = out
            errorChannel = err
        }

        project {
            inputPaths = sourcesFiles.map { it.toPath() }
        }

        baseline {
            shouldCreateDuringAnalysis = false
        }

        config {
            useDefaultConfig = true
            configPaths = listOf(File("path/to/detekt-config.yml").toPath())
        }

        execution {
            parallelParsing = false
            parallelAnalysis = false
        }

        rules {
            autoCorrect = false
            maxIssuePolicy = RulesSpec.MaxIssuePolicy.AllowAny
            activateAllRules = true
        }
    }
```

I've modified the options a few times, providing a config yaml file as well, haven't seen any changes to issue output.

When I call `DetektProvider.load().get(processingSpecHere).run()` and debug the application I find the following things:

Inside `Lifecycle` internal interface, the function `analyze` is called, it has the correct `filesToAnalyze`, the `processors` list contains 11 processors (`KtFileCount`, `PackageCount`, `ClassCount`, ...), the  the `ruleSets` contain only 1 item, which is `PerformanceProvider` (I think this may be the issue?).

The returned `Detektion` object contains 0 finding items (should have at least 1 `ImplicitDefaultLocale`), 0 notification items,. 5 metric items, 6 userdata items

That is for the first 2 items, there are about 309 files I pass to Detekt.

Thanks for any help :)

## Context
Trying to run Detekt on a codebase

## Your Environment
* Version of detekt used: 1.20.0
* Operating System and version: Mac OS

**Comment by cortinico** [comment]
> but when I am running `detekt` using the library `io.gitlab.arturbosch.detekt.api` I get 0 issues.

Could you clarify how you're using Detekt? Are you depending on the Detetk Jar and running it as part of a Java/Kotlin application? Can I ask you why you're doing this?

As a rule of thumb, you should be able to achieve the same results between the CLI and calling detekt from the .api package. That's how we're implementing the Gradle Plugin.

**Reply by manwinder123** [reply]
Using the jar for our Bazel project, running it as a Kotlin application. The application gathering issues in the codebase and mapping the issues into a JSON object which I can push to a server for visualization. I feel like I've setup the api package correctly, I didn't find much info on it though the actual API's are the exact same as the ones in the getting started tutorials. Any ideas on what part of my setup may be incorrect?

**Reply by manwinder123** [reply]
Also curious why `io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider` is the only ruleset

**Reply by manwinder123** [reply]
Ran my bazel app via terminal rather than Android Studio, debug mode enabled, here was the output:

```
Loaded extensions: 
    io.github.detekt.metrics.processors.ProjectSLOCProcessor@4ce25e47
    io.github.detekt.metrics.processors.ProjectLOCProcessor@6380e9e9
    io.github.detekt.metrics.processors.ProjectCLOCProcessor@350da119
    io.github.detekt.metrics.processors.ProjectLLOCProcessor@6a09484c
    io.github.detekt.metrics.processors.ProjectCognitiveComplexityProcessor@34c7232c
    io.github.detekt.metrics.processors.ProjectComplexityProcessor@50ec4bfc
    io.github.detekt.metrics.processors.PropertyCountProcessor@8b13d91
    io.github.detekt.metrics.processors.FunctionCountProcessor@285a4fe3
    io.github.detekt.metrics.processors.ClassCountProcessor@2589d787
    io.github.detekt.metrics.processors.PackageCountProcessor@1788cb61
    io.github.detekt.metrics.processors.KtFileCountProcessor@16d431b4
Registered rule sets: 
    io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@12abcd1e
Loaded extensions: 
    io.gitlab.arturbosch.detekt.core.baseline.BaselineResultMapping@2ef63b6d
Loaded extensions: 
    io.github.detekt.report.xml.XmlOutputReport@ad6255e
    io.github.detekt.report.txt.TxtOutputReport@81a56b2
    io.github.detekt.report.sarif.SarifOutputReport@488c6c98
    io.github.detekt.report.html.HtmlOutputReport@2530baf6
Loaded extensions: 
    io.gitlab.arturbosch.detekt.core.reporting.console.LiteFindingsReport@620e0d36
Phase LoadConfig took PT0.078S
Phase CreateSettings took PT0.01S
Phase ValidateConfig took PT0.002S
Phase Parsing took PT1.32S
Phase Binding took PT0.005S
Phase LoadingExtensions took PT0.01S
Phase Analyzer took PT0.832S
Phase Reporting took PT0.007S
```

**Reply by cortinico** [reply]
> Using the jar for our Bazel project

Have you looked into how https://github.com/buildfoundation/bazel_rules_detekt/ is using Detekt?

**Reply by manwinder123** [reply]
I was able to get more registered rule sets working by adding to my bazel build file, adding dependencies such as: `io.gitlab.arturbosch.detekt:detekt-report-xml:jar:1.20.0`, but the my custom RuleSetProvider doesn't get called or used, same with `DetektProvider`, I am using the following file names: 

`io.github.detekt.tooling.api.DetektProvider`
`io.gitlab.arturbosch.detekt.api.RuleSetProvider`

also have tried putting them in resources folder and have them outside as that's how `io.gitlab.arturbosch.detekt:detekt-report-xml:jar:1.20.0` had it

**Reply by manwinder123** [reply]
Figured out that I needed bazel to know I had resources, added this to my bazel BUILD file: 

```
    resources = [
        ":src/main/resources/META-INF/services/io.github.detekt.tooling.api.DetektProvider",
        ":src/main/resources/META-INF/services/io.gitlab.arturbosch.detekt.api.RuleSetProvider"
    ],
``` 

And Detekt started picking up issues and my custom rules
